mlrun/api/crud/artifacts.py

```diff
@@ -23,6 +23,15 @@ def store_artifact(
         project: str = mlrun.mlconf.default_project,
     ):
         project = project or mlrun.mlconf.default_project
+        # In case project is an empty string the setdefault won't catch it
+        if not data.setdefault("project", project):
+            data["project"] = project
```

Comment:
If you handled the case of an empty string, I suggest that you'll cover it in the tests (maybe I missed it ?)

Reply:
added unittests